src/actions/expressions.js

```diff
@@ -97,12 +98,26 @@ export function evaluateExpressions(frameId: frameIdType) {
 }
 
 function evaluateExpression(expression: Expression, frameId: frameIdType) {
-  return function({ dispatch, getState, client }: ThunkArgs) {
+  return async function({ dispatch, getState, client }: ThunkArgs) {
     if (!expression.input) {
       console.warn("Expressions should not be empty");
       return;
     }
 
+    const error = await parser.hasSyntaxError(expression.input);
```

Comment:
@jasonLaster i like `hasSyntaxError`!

Reply:
Me too.
I always discuss that sort of naming with my team mates.
If something is named validateFoo I'd expect to get a validation result which provides detailed infos of that validation. Otherwise I'd name it can/has/is Fooable

Reply:
:+1: for is/has for functions returning booleans.